symphony-bdk-core/src/main/java/com/symphony/bdk/core/extension/BdkAuthenticationAware.java

```diff
@@ -0,0 +1,11 @@
+package com.symphony.bdk.core.extension;
+
+import com.symphony.bdk.core.auth.AuthSession;
+
+import org.apiguardian.api.API;
+
+@API(status = API.Status.EXPERIMENTAL)
+public interface BdkAuthenticationAware {
+
+  void setAuthSession(AuthSession session);
```

Comment:
Correct, I actually started to move the Retry logic to a separate module but it would bring quite a lot of additional changes that will make this PR bigger. I'd prefer to do it as part of another PR once this one merged